desktop/core/src/desktop/js/reactComponents/Table/Table.tsx

```diff
@@ -0,0 +1,157 @@
+  onRowSelect?: (selectedRows: T[]) => void;
+  scroll?: RcTableProps<T>['scroll'];
+  sortByColumn?: ColumnProps<T>['dataIndex'];
+  sortOrder?: SortOrder;
+  setSortByColumn?: (column: ColumnProps<T>['dataIndex']) => void;
+  setSortOrder?: (order: SortOrder) => void;
+  testId?: string;
+  rowKey: ((record: T) => string) | string;
+  pagination?: PaginationProps;
+  rowClassName?: ((record: T) => string) | string;
+}
+
+function Table<T extends object>({
+  title,
+  data,
+  columns,
+  onRowClick,
+  onRowSelect,
+  scroll,
+  sortByColumn,
+  sortOrder,
+  setSortByColumn,
+  setSortOrder,
+  pagination,
+  testId,
+  locale,
+  rowKey,
+  rowClassName
+}: TableProps<T>): JSX.Element {
+  const rowSelection = onRowSelect
+    ? {
+        hideSelectAll: !data.length,
+        columnWidth: 36,
+        type: 'checkbox' as RowSelectionType,
+        onChange: (_: React.Key[], selectedRows: T[]) => {
+          onRowSelect(selectedRows);
+        }
+      }
+    : undefined;
+
+  const onColumnClick = (columnClicked: ColumnProps<T>['dataIndex']) => {
+    const isSortable = columns.find(col => col.dataIndex === columnClicked)?.sorter;
+    if (!isSortable || !setSortByColumn || !setSortOrder) {
+      return;
+    }
+
+    if (columnClicked === sortByColumn) {
+      const nextOrder = {
+        [SortOrder.NONE]: SortOrder.ASC,
+        [SortOrder.ASC]: SortOrder.DSC,
+        [SortOrder.DSC]: SortOrder.NONE
+      }[sortOrder as SortOrder];
+      setSortOrder(nextOrder);
+      if (nextOrder === SortOrder.NONE) {
+        setSortByColumn(undefined);
+      }
+    } else {
+      setSortByColumn(columnClicked);
+      setSortOrder(SortOrder.ASC);
+    }
+  };
+
+  const getColumnsFromConfig = (columnsConfig: ColumnProps<T>[]) => {
+    return columnsConfig.map(col => ({
+      ...col,
+      sorter: false, // we have custom sorter
+      title: (
+        <div
+          className={`hue-table__header ${col.sorter ? 'hue-table__header--cursor-pointer' : ''}`}
+          onClick={() => {
+            if (col.sorter) {
+              onColumnClick(col.dataIndex);
+            }
+          }}
+        >
+          {typeof col.title === 'function' ? col.title({}) : col.title}
+          {col.dataIndex === sortByColumn && (
+            <>
+              {sortOrder === SortOrder.DSC && <SortDescendingIcon />}
+              {sortOrder === SortOrder.ASC && <SortAscendingIcon />}
+            </>
+          )}
+        </div>
+      )
+    }));
+  };
+
+  return (
+    <>
+      <AntdTable
+        title={title}
+        className="hue-table"
+        columns={getColumnsFromConfig(columns)}
+        dataSource={data}
+        onRow={onRowClick}
+        pagination={false}
+        rowClassName={rowClassName}
+        rowKey={rowKey}
+        rowSelection={rowSelection}
+        scroll={scroll}
+        data-testid={testId}
+        locale={locale}
+        sticky
+      />
+      {pagination?.pageStats && pagination?.pageStats?.totalPages > 0 && (
```

Comment:
I see, so this is an "complex" table component that has pagination built in to it. Perhaps we should call it "PaginatedTable" then? For the chat app I built an "InfiniteScrollTable" that we might include in the near future,